lib/utils/constraints-utils.ts

```diff
@@ -0,0 +1,82 @@
+import { z } from "zod";
+
+export interface Constraint {
+    validate(key: string, value: any, identifier: string): any;
+}
+
+
+export class StringConstraint implements Constraint {
+
+    constructor(readonly min?: number, readonly max?: number) { }
+
+    validate(key: string, value: any, identifier: string) {
+
+        if (value != undefined)
+            z.string()
+                .min(this.min ?? 0, { message: key + " (" + identifier + ": " + value + ") must be no less than " + this.min + " characters long!" })
+                .max(this.max ?? 63, { message: key + " (" + identifier + ": " + value + ") must be no more than " + this.max + " characters long!" })
+                .parse(value);
+    }
+}
+
+export class UrlStringConstraint implements StringConstraint{
+
+    constructor(readonly min?: number, readonly max?: number) { }
+
+    validate(key: string, value: any, identifier: string) {
+
+        if (value != undefined){
+
+            z.string().url({ message: key + " (" + identifier + ": " + value + ") must be a URL!"}).parse(value);
+
+            z.string()
+                .min(this.min ?? 0, { message: key + " (" + identifier + ": " + value + ") must be a URL no less than " + this.min + " characters long!" })
+                .max(this.max ?? 100, { message: key + " (" + identifier + ": " + value + ") must be a URL no more than " + this.max + " characters long!" })
+                .parse(value);
+        }
+    }
+}
+
+export class NumberConstraint implements Constraint {
+
+    constructor(readonly min?: number, readonly max?: number) { }
+
+    validate(key: string, value: any, identifier: string) {
+
+        if (value != undefined)
+            z.number()
+                .gte(this.min ?? 1, { message: key + " (" + identifier + ": " + value + ") must be no less than " + this.min + " nodes!" })
+                .lte(this.max ?? 3, { message: key + " (" + identifier + ": " + value + ") must be no more than " + this.max + " nodes!" })
+                .parse(value);
+    }
+}
+
+export class ArrayConstraint implements Constraint {
+
+    constructor(readonly min?: number, readonly max?: number) { }
+
+    validate(key: string, value: any, identifier: string) {
+
+        if (value != undefined)
+            z.number()
+                .gte(this.min ?? 1, { message: key + " (" + identifier + ": " + value + ") must be no less than " + this.min + " node groups!" })
+                .lte(this.max ?? 3, { message: key + " (" + identifier + ": " + value + ") must be no more than " + this.max + " node groups!" })
+                .parse(value.length);
+    }
+}
+
+export type ConstraintsType<T> = Partial<Record<keyof T, Constraint>>;
+
+export function validateConstraints<T>(object: any, constraints: ConstraintsType<T>, context: string) {
```

Comment:
You can use a multi-arg support here similar to what we do [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/stacks/eks-blueprint-stack.ts#L116)

This enables you to pass either individual object or an array. The only caveat is  that ...arg must be last in the declaration.